src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -40,6 +40,28 @@ public override NoPia.EmbeddedTypesManager EmbeddedTypesManagerOpt
         /// </summary>
         private Dictionary<FieldSymbol, NamedTypeSymbol> _fixedImplementationTypes;
 
+        private bool _needsGeneratedIsReadOnlyAttribute_IsFrozen;
+        private bool _needsGeneratedIsReadOnlyAttribute_Value;
```

Comment:
Discussed offline. We cannot use it if we want to always throw after freezing.

---
In reply to: [115099940](https://github.com/dotnet/roslyn/pull/18715#discussion_r115099940) [](ancestors = 115099940,115029940)